packages/@react-spectrum/combobox/stories/ComboBox.stories.tsx

```diff
@@ -411,7 +420,148 @@ storiesOf('ComboBox', module)
     () => (
       <CustomFilterComboBox />
     )
+    )
+  .add(
+    'isLoading',
+    () => (
+      <ComboBox label="Combobox" isLoading defaultItems={[]}>
+        {item => <Item>{item.name}</Item>}
+      </ComboBox>
+    )
+  )
+  .add(
+    'isLoading, validationState: invalid',
+    () => (
+      <ComboBox label="Combobox" isLoading validationState="invalid" defaultItems={[]}>
+        {item => <Item>{item.name}</Item>}
+      </ComboBox>
+    )
+  )
+  .add(
+    'isLoading, isQuiet',
+    () => (
+      <ComboBox label="Combobox" isLoading isQuiet defaultItems={[]}>
+        {item => <Item>{item.name}</Item>}
+      </ComboBox>
+    )
+  )
+  .add(
+    'isLoading, isQuiet, validationState: invalid',
+    () => (
+      <ComboBox label="Combobox" isLoading isQuiet validationState="invalid" defaultItems={[]}>
+        {item => <Item>{item.name}</Item>}
+      </ComboBox>
+    )
+  )
+  .add(
+    'filtering with useListData',
+    () => (
+      <ListDataExample />
+    )
+  )
+  // TODO: test pagination via scrolling with this
+  .add(
+    'client side filtering with useAsyncList',
+    () => (
+      <AsyncLoadingClientExample />
+    )
+  )
+  .add(
+    'server side filtering with useAsyncList',
+    () => (
+      <AsyncLoadingExample />
+    )
+  );
+
+function ListDataExample() {
+  let {contains} = useFilter({sensitivity: 'base'});
+  let list = useListData({
+    initialItems: items,
+    filterFn(item, text) {
+      return contains(item.name, text);
+    }
+  });
+
+  return (
+    <ComboBox
+      label="ComboBox"
+      items={list.items}
+      inputValue={list.filterText}
+      onInputChange={list.setFilterText}>
+      {item => <Item>{item.name}</Item>}
+    </ComboBox>
+  );
+}
+
+function AsyncLoadingClientExample() {
+  interface Pokemon {
+    name: string,
+    url: string
+  }
+
+  let {contains} = useFilter({sensitivity: 'base'});
+
+  let list = useAsyncList<Pokemon>({
+    async load({signal, cursor}) {
+      let res = await fetch(cursor || 'https://pokeapi.co/api/v2/pokemon', {signal});
+      let json = await res.json();
+      // The API is too fast sometimes, so make it take longer so we can see the spinner
+      await new Promise(resolve => setTimeout(resolve, cursor ? 500 : 1000));
+
+      return {
+        items: json.results,
+        cursor: json.next
+      };
+    },
+    filterFn(item, text) {
+      return contains(item.name, text);
+    },
+    initialFilterText: 'B'
+  });
+
+  return (
+    <ComboBox
+      label="Pick a Pokemon"
+      items={list.items}
+      inputValue={list.filterText}
+      onInputChange={list.setFilterText}
+      isLoading={list.isLoading}
+      onLoadMore={list.loadMore}>
+      {item => <Item key={item.name}>{item.name}</Item>}
+    </ComboBox>
+  );
+}
+
+
+function AsyncLoadingExample() {
```

Comment:
Note that this example will close the menu every time the combobox text is changed. IMO this feels kinda weird, perhaps I should change it so the dropdown stays open and the contents just replaced with a loading swirl